mpbb-install-dependencies

```diff
@@ -101,34 +101,65 @@ install-dependencies() {
         text="Installing dependency ($dependencies_counter of $dependencies_count) '${depname}' with variants '${depvariants}'"
         echo "----> ${text}"
         echo -n "${text} ... " >> "$log_status_dependencies"
-        # $depvariants isn't quoted on purpose
-        # shellcheck disable=SC2086
-        if ! "${option_prefix}/bin/port" -d install --unrequested "$depname" $depvariants; then
-            echo "Build of dependency '${depname}' with variants '${depvariants}' failed, aborting." >&2
-            echo "[FAIL]" >> "$log_status_dependencies"
-            echo "Building '$port' ... [ERROR] (failed to install dependency '${depname}') maintainers: $(get-maintainers "$port" "${depname}")." >> "$log_subports_progress"
-
-            # Update failcache
-            # $depvariants isn't quoted on purpose
-            # shellcheck disable=SC2086
-            failcache_failure "$depname" $depvariants
-            if [ $? -ne 0 ]; then
-                err "failcache_failure $depname $depvariants failed."
-                return 1
-            fi
-            return 1
-        else
+        # $option_prefix and $thisdir are set in mpbb
+        # shellcheck disable=SC2154
+        if [[ -f $("${option_prefix}/bin/port-tclsh" "${thisdir}/tools/archive-path.tcl" "${depname}" "${depvariants}") ]]; then
```

Comment:
This part looks a little bit weird to me. In my opinion we should not even run `install-dependencies` when the package exists. On the other hand, when we actually need dependencies for a new port, this step will skip the required activation and most likely postpone it to the "install" step of the port itself, which is where we don't want the activation logs of dependencies.

If anything, you should probably test for existence of the port rather than for existence of dependencies. And then skip the whole chunk of installing dependencies rather that counting existing ones.